src/Sarif.Sarifer/TextBufferViewTracker.cs

```diff
@@ -43,7 +43,7 @@ public void AddTextView(ITextView textView, string path, string text)
             trackingInformation.Add(textView);
             if (first)
             {
-                FirstViewAdded?.Invoke(this, new FirstViewAddedEventArgs(path, text));
+                FirstViewAdded?.Invoke(this, new FirstViewAddedEventArgs(path, text, trackingInformation.CancellationToken));
```

Comment:
>, trackingInformation.CancellationToken) [](start = 83, length = 40)

It's not standard to pass a cancellation token in an event.. I would remove it here.

Reply:
On second thought. :) If this event is used internally to track views :) I would pass the whole trackingInformation.

---
In reply to: [524853366](https://github.com/microsoft/sarif-visualstudio-extension/pull/274#discussion_r524853366) [](ancestors = 524853366)